backend/database/util.py

```diff
@@ -19,7 +20,30 @@ def create_user(userdata: dict):
     except Exception as e:
         db.session.rollback()
         return False
+    
+def update_user(email, userdata: dict):
+    try:
+        # with db.session.begin(subtransactions=True):
+        user = User.query.filter_by(email=email).first()
+        if user:
+            for key, value in userdata.items():
+                setattr(user, key, value)
+        db.session.commit()
+        return True
+    except Exception as e:
+        db.session.rollback()
+        print(e)
+        return False 
```

Comment:
**_MANDATORY:_**
I rather send the user itself everytime we update. I got it working on my branch. Could you just use the following instead? I have test already.
```suggestion
def update_user(steam_id, new_fields):
    try:
        user = User.query.filter_by(steam_id=steam_id).first()
        if user:
            for key, value in new_fields.items():
                setattr(user, key, value)
            db.session.commit()
            return user.to_dict()
        else:
            return None
    except Exception as e:
        db.session.rollback()
        return e
```